frontend/src/components/SignUp/signup.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import { Link } from "react-router-dom";
+import styles from'./signup.module.css';
+import gmail_icon from '../Logos/gmail_icon.png';
+import fb_icon from '../Logos/fb_icon.png';
+import twtx_icon from '../Logos/twtx_icon.webp';
+import apple_icon from '../Logos/apple_icon.png';
+
+function EmailErrorMessage({username, isValid}) {
+    return <p className={styles.error_message}>Please enter a valid email.</p>
+}
+
+function PasswordErrorMessage({password, isValid}) {
+    return <p className={styles.error_message}>Passwords must be at least 8 characters and 
+    include at least one lowercase, uppercase, number, and special symbol (~|!@#$%^&*).</p>
+}
+
+function PasswordConfirmation({password, isValid}) {
+    return <p className={styles.error_message}>Passwords do not match. </p>
+}
+
+export function SignUp() {
+
+    // notes:
```

Comment:
Since we are going to be merging this code into main we don't want a big chunk of comments in our codebase. I'd highly recommend just making a google doc to put down any notes you need for the future. But please remove these comments for the PR.